README.md

```diff
@@ -440,16 +430,78 @@ Those SNAPSHOTS were built and deployed last night by Jenkins jobs.
+    - `build`, build functionality.
+    - `branch -b <br>`, checkout projects in according to their branch starting from `<br>` branch in `<starting-project>` project.
+    - `--fullProjectDependencyTree`, checks out and execute the whole tree instead of the upstream build, if omitted the upstream flow is used.
+    - `-sp <starting-project>`, from which project (in the tree structure) start from.
+    - `--skipExecution`, add this if you only want to clone all repositories without building them.
+    
+    A real example could be this one, where you checkout and build the whole set of projects starting from the root one (i.e., `kiegroup/droolsjbpm-build-bootstrap`):
+
+    ```shell
+    $ build-chain-action -df 'https://raw.githubusercontent.com/kiegroup/droolsjbpm-build-bootstrap/main/.ci/pull-request-config.yaml' build branch -b main --fullProjectDependencyTree -sp kiegroup/droolsjbpm-build-bootstrap
+    ```
+
+* **Pull Request Flow**, build projects related to a specific Pull Request you are performing:
+
+    ``` shell
+    $ build-chain-action -df <definition-file> build pr -url <pull-request-url> [-sp <starting-project>] [--skipExecution]
+    ```
+
+    Consider the scenario where you have multiple PRs, related to the same topic, on different RHBA projects. With this command you are able to checkout all of them in the correct PR branch and build all of them. This will allow you to build and test the overall scenario having multiple PR-related projects.
+    
+    - `-df <definition-file>`, url or path to the build-chain definition file (more details [here](https://github.com/kiegroup/build-chain-configuration-reader)).
+    - `build`, build functionality.
+    - `pr -url <pull-request-url>`, for every projects checks if a Pull Request exists (having the same name of the provided one) and if yes it check outs in that branch.
+    - `-sp <starting-project>`, from which project (in the tree structure) start from, if not provided it starts from the PR one.
+    - `--skipExecution`, add this if you only want to clone all repositories without building them.
+
+    A real example could be this one, taken from `kie-wb-distribution` project:
+
+    ```shell
+    $ build-chain-action -df 'https://raw.githubusercontent.com/${GROUP:kiegroup}/droolsjbpm-build-bootstrap/${BRANCH:main}/.ci/pull-request-config.yaml' build pr -url https://github.com/kiegroup/kie-wb-distributions/pull/1166 
+    ```
+
 Running tests
 -------------
 
-* All modules
+* Single project
 
     ```shell
     $ cd ~/projects/kiegroup/drools
     $ mvn test [-Dtest=ATestClassName]
     ```
 
+
+* Set of projects
+    
+    This can be reached using the `build-chain-action` tool, introduced in [Running the Build](#running-the-build) section. Omitting the `--skipExecution` flag you will check out all projects and build them (also executing all tests).
```

Comment:
```suggestion
    This can be reached using the `build-chain` tool, introduced in [Running the Build](#running-the-build) section. Omitting the `--skipExecution` flag it will check out and build the whole set of RHBA projects (also executing all tests).
```